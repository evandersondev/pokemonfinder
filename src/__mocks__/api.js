export default [
  {
    id: 1,
    name: 'Pokemon 1',
    sprites: {
      front_default:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png',
      back_default:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png',
    },
    abilities: [
      { slot: 1, ability: { name: 'ability 1' } },
      { slot: 1, ability: { name: 'ability 1' } },
    ],
    height: 15,
    weight: 90,
    stats: [
      { base_stat: 90, stat: { name: 'stat 1' } },
      { base_stat: 240, stat: { name: 'stat 2' } },
    ],
    types: [{ type: { name: 'type 1' } }, { type: { name: 'type 2' } }],
  },
  {
    id: 2,
    name: 'Pokemon 2',
    sprites: {
      front_default:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png',
      back_default:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png',
    },
    abilities: [
      { slot: 1, ability: { name: 'ability 1' } },
      { slot: 1, ability: { name: 'ability 1' } },
    ],
    height: 15,
    weight: 90,
    stats: [
      { base_stat: 90, stat: { name: 'stat 1' } },
      { base_stat: 240, stat: { name: 'stat 2' } },
    ],
    types: [{ type: { name: 'type 1' } }, { type: { name: 'type 2' } }],
  },
  {
    id: 3,
    name: 'Pokemon 3',
    sprites: {
      front_default:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png',
      back_default:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png',
    },
    abilities: [
      { slot: 1, ability: { name: 'ability 1' } },
      { slot: 1, ability: { name: 'ability 1' } },
    ],
    height: 15,
    weight: 90,
    stats: [
      { base_stat: 90, stat: { name: 'stat 1' } },
      { base_stat: 240, stat: { name: 'stat 2' } },
    ],
    types: [{ type: { name: 'type 1' } }, { type: { name: 'type 2' } }],
  },
];
