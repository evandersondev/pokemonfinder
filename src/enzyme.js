import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });
export { shallow };
