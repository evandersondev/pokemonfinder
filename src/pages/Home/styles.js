import styled from 'styled-components';

export const Container = styled.div`
  width: 90%;
  max-width: 920px;
  margin: auto;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;

export const Pagination = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  margin-bottom: 30px;

  a {
    width: 120px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    background: transparent;
    border: 2px solid #f6e58d;
    border-radius: 3px;
    color: #f6e58d;
    font-size: 10px;
    cursor: pointer;
    transition: all 0.3s ease-in-out;

    &:hover {
      background: #f6e58d;
      color: #333333;
    }

    &[disabled] {
      opacity: 0.2;
      color: #333333;
      border-color: #f6e58d;
      background: #f6e58d;
      cursor: default;
      transition: none;
      pointer-events: none;
    }
  }
`;
