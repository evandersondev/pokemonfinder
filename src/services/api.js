import { create } from 'axios';

const pokeApi = create({
  baseURL: 'https://pokeapi.co/api/v2',
});

export default {
  async getAllPaginated({ perPage, page = 1 }) {
    const offset = perPage * (page - 1);

    const {
      data: { results, previous, next },
    } = await pokeApi.get('/pokemon', {
      params: {
        offset,
        limit: perPage,
      },
    });

    return { results, previous, next };
  },

  async getPokemonByName({ name }) {
    const { data } = await pokeApi.get(`pokemon/${name}`);

    return data;
  },
};
