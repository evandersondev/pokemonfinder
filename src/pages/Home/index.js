import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import queryString from 'query-string';

import api from '../../services/api';
import { Container, Pagination } from './styles';

import Card from '../../components/Card';
import Loading from '../../components/Loading';

export default () => {
  const [pokemons, setPokemons] = useState([]);
  const [pagination, setPagination] = useState({});
  const [loading, setloading] = useState(false);

  const query = queryString.parse(window.location.search);
  const page = parseInt(query.page || 1);
  const perPage = 21;

  const fetchPokemon = async () => {
    setloading(true);

    const { results, previous, next } = await api.getAllPaginated({
      perPage,
      page,
    });

    setPokemons(results);
    setPagination({ previous, next });
    setloading(false);
  };

  useEffect(() => {
    fetchPokemon();
  }, [page]);

  return (
    <Container>
      {loading ? (
        <Loading />
      ) : (
        <>
          {pokemons &&
            pokemons.map((pokemon) => (
              <Card className="card" key={pokemon.name} pokemon={pokemon} />
            ))}

          <Pagination>
            <Link disabled={pagination.previous} to={`?page=${page - 1}`}>
              Previous
            </Link>

            <Link disabled={pagination.next} to={`?page=${page + 1}`}>
              Next
            </Link>
          </Pagination>
        </>
      )}
    </Container>
  );
};
