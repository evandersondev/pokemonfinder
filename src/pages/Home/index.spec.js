/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import api from '../../services/api';

import Home from '.';

jest.mock('../../services/api', () => {
  return {
    getAllPaginated({ perPage, page = 1 }) {
      return {
        results: [{ name: 'spearow' }],
        previous: null,
        next: 'url/qualquer',
      };
    },
  };
});

describe('test render components in Home', () => {
  it('renders corretly', () => {
    const wrapper = shallow(<Home />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should have two links', () => {
    const wrapper = shallow(<Home />);

    expect(wrapper.find('Link').length).toBe(2);
  });

  it('should a Link with name Previous and Next', () => {
    const wrapper = shallow(<Home />);

    const text = wrapper.find('Link').map((node) => node.text());
    expect(text).toEqual(['Previous', 'Next']);
  });
});
