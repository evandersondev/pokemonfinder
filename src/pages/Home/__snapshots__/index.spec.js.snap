// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render components in Home renders corretly 1`] = `
<styled.div>
  <styled.div>
    <Link
      disabled={false}
      to="?page=0"
    >
      Previous
    </Link>
    <Link
      disabled={false}
      to="?page=2"
    >
      Next
    </Link>
  </styled.div>
</styled.div>
`;
